src/sentry/search/eap/resolver.py

```diff
@@ -391,11 +391,44 @@ def resolve_term(
     ) -> tuple[TraceItemFilter, VirtualColumnDefinition | None]:
         resolved_column, context_definition = self.resolve_column(term.key.name)
 
+        value = term.value.value
+        # time series request do not support virtual column contexts, so we have to remap the value back to the original column. (see https://github.com/getsentry/eap-planning/issues/236)
+        if self.params.is_timeseries_request and context_definition is not None:
```

Comment:
lets move this logic into a helper function on the SearchResolver class for readability (then the comment can be a docstring on that helper too)